subprojects/core/src/main/java/org/gradle/internal/classpath/types/GradleCoreInstrumentingTypeRegistry.java

```diff
@@ -90,11 +94,11 @@ private static Optional<HashCode> loadUpgradedPropertiesHash() {
 
     private static Optional<HashCode> loadHashCodeFromResource(String resourceFile) {
         try (InputStream stream = GradleCoreInstrumentingTypeRegistry.class.getResourceAsStream(resourceFile)) {
-            byte[] bytes = IOUtils.toByteArray(stream);
-            if (bytes.length == 0) {
+            if (stream == null) {
                 return Optional.empty();
             }
-            return Optional.of(HashCode.fromBytes(bytes));
+            StreamHasher streamHasher = new DefaultStreamHasher();
```

Comment:
:x: This should be injected as a service.

Reply:
Fixed